lib/rules/define-props-declaration.js

```diff
@@ -6,6 +6,168 @@
 
 const utils = require('../utils')
 
+const mapNativeType = (/** @type {string} */ nativeType) => {
+  switch (nativeType) {
+    case 'String': {
+      return 'string'
+    }
+    case 'Number': {
+      return 'number'
+    }
+    case 'Boolean': {
+      return 'boolean'
+    }
+    case 'Object': {
+      return 'Record<string, any>'
+    }
+    case 'Array': {
+      return 'any[]'
+    }
+    case 'Function': {
+      return '() => void'
```

Comment:
I think all functions should be allowed by default, so we have to allow any number of arguments and any return type:

```suggestion
      return '(...args: any[]) => any'
```